xarray/core/computation.py

```diff
@@ -1825,11 +1825,10 @@ def where(cond, x, y, keep_attrs=None):
     """
     if keep_attrs is None:
         keep_attrs = _get_keep_attrs(default=False)
-
     if keep_attrs is True:
         # keep the attributes of x, the second parameter, by default to
         # be consistent with the `where` method of `DataArray` and `Dataset`
-        keep_attrs = lambda attrs, context: attrs[1]
+        keep_attrs = lambda attrs, context: getattr(x, "attrs", {})
```

Comment:
What is `attrs` here for a scalar? The issue is that you're using `x.attrs` which could be a dataset while presumably `attrs` has Variable attributes when `apply_ufunc` is iterating through variables in a dataset.. We want `attrs[1] or {}` I think with some handling for errors.

PS: Sorry for the not very throrough review earlier

Reply:
I think something like this should work (with `getitem` defined in `xarray.core.utils`):
```python
_default = object()
def getitem(sequence, item, default=_default):
    try:
        return sequence[item]
    except IndexError:
        if default is _default:
            raise
        return default

keep_attrs = lambda attrs, context: getitem(attrs, 1, {})
```